QuestionAnswering/QA_ICL/backend/fastapi_app/README.md

```diff
@@ -116,6 +121,308 @@ Overview:
+   deactivate entity_store
+
+   controller->>executor: Forwards data request to execution engine
+   activate executor
+   executor->>controller: Returns execution results
+   deactivate executor
+
+   controller->>vis_data_store: Requests data for visualisation
+   activate vis_data_store
+   vis_data_store->>controller: Returns data for visualisation
+   deactivate vis_data_store
+
+   controller->>client: Returns data for answering<br/>and visualisation
+   deactivate controller
+```
+
+### Entity Linking
+The following sequence diagram gives a closer look at the activity happens before and after entity linking, which gives a better understanding on the input and use of output of entity linking.
+```mermaid
+sequenceDiagram
+    participant qa as DataSupporter
+    participant llm_caller as Nlq2DataReqLLMCaller
+    participant entity_store as CentralEntityLinker
+    participant executor as DataReqExecutor
+    participant sparql_executor as SparqlDataReqExecutor
+    participant func_executor as FuncDataReqExecutor
+
+    llm_caller ->> qa: generate structure data request <br/> and output JSON object NLQ to Data Req
+    Note over llm_caller,qa: Sample NLQ2DataReq JSON can be found below
+
+    loop for each Entity
+        loop for each Entity value
+            qa ->> entity_store: link entity with entity class (var2class) and entity value
+            entity_store ->> qa: entity value IRI
+        end
+    end
+
+    qa ->> executor: execute query with var2cls, var2iris
+    alt is Sparql req
+        executor ->> sparql_executor: execute sparql query
+        sparql_executor ->> executor: support_data, data_artifact, vis_var2iris
+    else is Func req
+        executor ->> func_executor: execute function call
+        func_executor ->> executor: data, artifact, vis_var2iris
+    end
+
+```
+
+Entity Linking sequence diagram
+
+```mermaid
+sequenceDiagram
+participant el as CentralEntityLinker
+participant redis as Redis
+participant lm as LinkerManager
+participant qa as DataSupporter
+participant triton as Triton
+
+el ->> el: check whether customized <br/> LinkerManager exists for the class
+alt exist
+    el ->> lm: link entity
+    lm ->> el: linked IRIs
+    el ->> qa: linked IRIs, end entity linking
+else not exists
+    el ->> el: read entity linking config<br/> of the class 
+    alt fuzzy
+        el ->> redis: exact match query
+        redis ->> el: exact matched IRIs
+        el ->> redis: fuzzy match query
+        radis ->> el: fuzzy matched IRIs
+        el ->> qa: all matched IRIs, end entity linking
+    else semantic
+        el ->> redis: exact match query
+        redis ->> el: exact matched IRIs
+        el ->> triton: embed entity value
+        triton ->> el: embedded entity value
+        el ->> redis: knn search of the embedded value
+        redis ->> el: matched IRIs
+        el ->> qa: matched IRIs, end entity linking
+    end
+end
+
+```
+
+#### Input & Output JSON Object Example
```

Comment:
I am a little unsure what the input and output are for? Is it for any specific routes or tasks? Is this an intermediate step? I have no context to this

Reply:
A clarification on the JSON example in FastAPI App README has been made.